modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/LocalizableRichText/LocalizableRichText.es.js

```diff
@@ -0,0 +1,190 @@
+	);
+	const [currentEditingLocale, setCurrentEditingLocale] = useState(
+		editingLocale
+	);
+	const [currentValue, setCurrentValue] = useState(
+		editable ? predefinedValue : value ?? predefinedValue
+	);
+	const [currentInternalValue, setCurrentInternalValue] = useState(
+		getInitialInternalValue({
+			editingLocale: currentEditingLocale,
+			value: currentValue,
+		})
+	);
+	useEffect(() => {
+		const editor = editorRef.current?.editor;
+		if (editor) {
+			editor.config.contentsLangDirection =
+				Liferay.Language.direction[currentEditingLocale.localeId];
+			editor.config.contentsLanguage = currentEditingLocale.localeId;
+			editor.setData(currentInternalValue);
+		}
+		const {availableLocales} = {
+			...transformAvailableLocalesAndValue({
+				availableLocales: currentAvailableLocales,
+				defaultLocale,
+				value: currentValue,
+			}),
+		};
+
+		setCurrentAvailableLocales(availableLocales);
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [currentEditingLocale]);
+
+	return (
+		<FieldBase
+			{...otherProps}
+			id={id}
+			name={name}
+			readOnly={readOnly}
+			style={readOnly ? {pointerEvents: 'none'} : null}
+			visible={visible}
+		>
+			<ClayInput.Group>
+				<ClayInput.GroupItem>
+					<ClassicEditor
+						className="w-100"
+						contents={currentValue[currentEditingLocale?.localeId]}
+						editorConfig={editorConfig}
+						name={name}
+						onBlur={onBlur}
+						onChange={(content) => {
+							if (
+								currentValue[currentEditingLocale?.localeId] !==
+								content
+							) {
+								const newValue = {
+									...currentValue,
+									[currentEditingLocale.localeId]: content,
+								};
+
+								setCurrentValue(newValue);
+								setCurrentInternalValue(content);
+
+								const {availableLocales} = {
+									...transformAvailableLocalesAndValue({
+										availableLocales: currentAvailableLocales,
+										defaultLocale,
+										value: newValue,
+									}),
+								};
+
+								setCurrentAvailableLocales(availableLocales);
+
+								onChange({target: {value: newValue}});
+							}
+						}}
+						onFocus={onFocus}
+						onSetData={({
+							data: {dataValue: value},
+							editor: {mode},
+						}) => {
+							if (mode === 'source') {
+								onChange({target: {value}});
+							}
+						}}
+						readOnly={readOnly}
+						ref={editorRef}
+					/>
+				</ClayInput.GroupItem>
+
+				<input
+					id={id}
+					name={name}
+					type="hidden"
+					value={currentValue || ''}
+				/>
+
+				<ClayInput.GroupItem
+					className="liferay-ddm-form-field-localizable-text"
+					shrink
+				>
+					<LocalesDropdown
+						availableLocales={currentAvailableLocales}
+						editingLocale={currentEditingLocale}
+						fieldName={fieldName}
+						onLanguageClicked={({localeId}) => {
+							const newEditingLocale = currentAvailableLocales.find(
+								(availableLocale) =>
+									availableLocale.localeId === localeId
+							);
+							setCurrentEditingLocale({
```

Comment:
@pedro-oliveira446 Please, add a blank line between the creation of a const and its usage.